fdbclient/include/fdbclient/MetaclusterManagement.actor.h

```diff
@@ -705,6 +709,12 @@ struct RemoveClusterImpl {
 		state DataClusterMetadata clusterMetadata = wait(getClusterTransaction(tr, self->clusterName));
 		wait(self->ctx.setCluster(tr, self->clusterName));
 
+		if ((self->clusterId.present() && clusterMetadata.entry.id != self->clusterId.get()) ||
+		    (!self->legalClusterStates.empty() &&
+		     !self->legalClusterStates.count(clusterMetadata.entry.clusterState))) {
+			throw cluster_already_exists();
```

Comment:
Would something like `cluster_removed` work better here? It's not intuitive to me why a cluster already existing would be an issue for a removal operation

Reply:
It turns out this matters very right now little because the only way to get here is through internal callers, and those internal callers are throwing away the error.

Essentially this is saying we cannot remove the cluster with the specified name and constraints because there is another cluster here with the same name. `cluster_removed` isn't terrible, but it's not necessarily a given that the cluster we are trying to match against was removed (for example it could have progressed to the ready state), and it's also a little weird for a removal operation to fail with an error saying the cluster was removed.

`cluster_already_exists` isn't necessarily the best option either since it's not completely intuitive what that means. I could create a new error, but it feels a little weird to create an error that is never used anywhere.

Maybe I could just use `operation_failed` and comment that the error is currently not used except to force the removal to fail. If that changes, we could adopt a new error at that time.